src/backend/feed/processor.js

```diff
@@ -186,6 +188,18 @@ module.exports = async function processor(job) {
     // Version info for this feed changed, so update the database
     feed.etag = feed.etag || info.etag;
     feed.lastModified = feed.lastModified || info.lastModified;
+    // If feed.link is undefined or empty add a link
+    if (!feed.link) {
+      const linkSet = new Set();
+      articles.forEach(article => {
+        // We only want to grab the link of the blog once
```

Comment:
The way you've written this, you don't need a loop or a Set, since you're essentially only grabbing the first link and skipping it on every other iteration.  Why not just use `articles[0].meta.link`?

Reply:
I was worried about the case someone has a blog but no posts yet. But, now that I think about it, I think we have something in Feed to check for null/undefined vals.